apps/server/src/modules/tool/mapper/external-tool-response.mapper.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@nestjs/common';
+import { ExternalToolResponse } from '@src/modules/tool/controller/dto/response/external-tool.response';
+import {
+	BasicToolConfigDO,
+	CustomParameterDO,
+	ExternalToolDO,
+	Lti11ToolConfigDO,
+	Oauth2ToolConfigDO,
+} from '@shared/domain/domainobject/external-tool';
+import { BasicToolConfigResponse } from '@src/modules/tool/controller/dto/response/basic-tool-config.response';
+import { Oauth2ToolConfigResponse } from '@src/modules/tool/controller/dto/response/oauth2-tool-config.response';
+import { CustomParameterResponse } from '@src/modules/tool/controller/dto/response/custom-parameter.response';
+import { Lti11ToolConfigResponse } from '../controller/dto/response/lti11-tool-config.response';
+
+@Injectable()
+export class ExternalToolResponseMapper {
+	mapToResponse(externalToolDO: ExternalToolDO): ExternalToolResponse {
+		let mappedConfig: BasicToolConfigDO | Lti11ToolConfigDO | Oauth2ToolConfigDO;
+		if (externalToolDO.config instanceof BasicToolConfigDO) {
+			mappedConfig = this.mapBasicToolConfigDO(externalToolDO.config);
+		} else if (externalToolDO.config instanceof Lti11ToolConfigDO) {
+			mappedConfig = this.mapLti11ToolConfigDO(externalToolDO.config);
+		} else {
+			mappedConfig = this.mapOauth2ToolConfigDO(externalToolDO.config);
+		}
+
+		const mappedCustomParameter: CustomParameterResponse[] = this.mapCustomParameterToResponse(
+			externalToolDO.parameters ?? []
+		);
+
+		return new ExternalToolResponse({
+			id: externalToolDO.id || '',
+			name: externalToolDO.name,
+			url: externalToolDO.url,
+			logoUrl: externalToolDO.logoUrl,
+			config: mappedConfig,
+			parameters: mappedCustomParameter,
+			isHidden: externalToolDO.isHidden,
+			openNewTab: externalToolDO.openNewTab,
+			version: externalToolDO.version,
+		});
+	}
+
+	private mapBasicToolConfigDO(externalToolConfigDO: BasicToolConfigDO): BasicToolConfigResponse {
```

Comment:
Here you use different namings. in line 56 you called it mapCustomParameterToResponse and here mapCustomParameterToResponse. I would prefer a consistent naming at least within classes

Reply:
done :)